go/consensus/tendermint/apps/staking/state/state_test.go

```diff
@@ -27,6 +32,85 @@ func mustInitQuantityP(t *testing.T, i int64) *quantity.Quantity {
 	return &q
 }
 
+func TestDelegationQueries(t *testing.T) {
+	db := dbm.NewMemDB()
+	tree := iavl.NewMutableTree(db, 128)
+	s := NewMutableState(tree)
+
+	fac := memorySigner.NewFactory()
+
+	// Generate escrow account.
+	escrowSigner, err := fac.Generate(signature.SignerEntity, rand.Reader)
+	require.NoError(t, err, "generating escrow signer")
+	escrowID := escrowSigner.Public()
+
+	escrowAccount := &staking.Account{}
+	s.SetAccount(escrowID, escrowAccount)
+
+	// Generate delegator accounts.
+	delegatorIDs := []signature.PublicKey{}
+	// Store expected delegations.
+	expectedDelegations := make(map[signature.PublicKey]map[signature.PublicKey]*staking.Delegation)
+	expectedDelegations[escrowID] = map[signature.PublicKey]*staking.Delegation{}
+	expectedDebDelegations := make(map[signature.PublicKey]map[signature.PublicKey][]*staking.DebondingDelegation)
+	expectedDebDelegations[escrowID] = map[signature.PublicKey][]*staking.DebondingDelegation{}
+
+	for i := int64(1); i <= numDelegatorAccounts; i++ {
+		signer, serr := fac.Generate(signature.SignerEntity, rand.Reader)
+		require.NoError(t, serr, "memory signer factory Generate account")
+		id := signer.Public()
+
+		delegatorIDs = append(delegatorIDs, id)
+
+		// Init account.
+		account := &staking.Account{}
+		account.General.Nonce = uint64(i)
+		require.NoError(t, account.General.Balance.FromBigInt(big.NewInt(2*i*100)), "initialize delegator account general balance")
```

Comment:
As a matter of taste, and (really old habit), I dislike assertions with side-effects.  Rewrite to separate setting the balance, and the assertion.

(Similar comments omitted for brevity.)

Reply:
fixed (throughout whole file)